### DISCUSSION TITLE
Blast campaign doesn't recalculate list size on send.

### DISCUSSION THREAD

**Discussion by williamhrs** [parcelvoy/platform#514] [General]
Hi @pushchris , I have a question for you.

when schedule a blast campaign, the list size get's calculated when we schedule it, but if the list size changes whenever it's going to be sent, the # of leads don't update. 

Do you think this is an error or you made it this way?

It kind of makes unpractical to schedule blasts for a long time, beucause our list size changes all the time because of people joining and people that don't open the e-mail for some time

**Comment by pushchris** [comment]
This is how it was created. When you schedule a blast it preemptively creates send records based on your list at the moment you schedule so that it can more rapidly process the send when the time comes. We'd have to figure out some other logic path for when to setup the list otherwise because it gets really tricky considering timezones and such to know when to make the cutoff for list generation especially when it can take a good amount of time to actually generate a list once you get into millions of users. The primary way the scheduling gets used is folks setting up a send for say the next day but they want it to arrive in all users inboxes at 9am regardless of their timezone.

**Reply by williamhrs** [reply]
I see! 

I've seen other tools that at least calculate the lists daily, would that be possible? Anyways @leobarcellos will take a look at it to see.

**Reply by pushchris** [reply]
That could definitely be possible! Would just need to setup some sort of scheduler that could regenerate it until the time of the first message starting to go out. I will have to checkout how the other tools out there handle this.

**Comment by pushchris** [comment]
Have started some initial work on changing this behavior, can hopefully finish testing and close out later this week if all goes well: https://github.com/parcelvoy/platform/pull/519

**Reply by williamhrs** [reply]
That's awesome!

**Comment by williamhrs** [comment]
@pushchris another thing about list calculation. 

On our operation, we want to only send emails to people that have opened our emails recently, are engaged with out content.

Today, we're updating daily our lists to reflect people that opened or clicked an emails for the past X days. For example: 10 days
To create that we do something like:

![image](https://github.com/user-attachments/assets/5867ae08-e465-4886-926f-ea2842d645bf)

It would be awesome to have that , having the abbility to the list (even if it's only updateing once a day ) to be always related to the last X days or something lke that. 

![image](https://github.com/user-attachments/assets/d88b56e4-1b15-4e54-9658-0e259f4fc973)

Does it make sense? 

Did I make myself clear enough? haha

**Reply by pushchris** [reply]
This functionality has been implemented here: https://github.com/parcelvoy/platform/pull/538 and you can find additional documentation here: https://docs.parcelvoy.com/how-to/lists#dynamic-lists

**Reply by williamhrs** [reply]
Awesome!
